Profiles

bananabr3d opened the pull request:
Manages profiles.json. Can Add, update and remove profiles

bananabr3d committed:
feat: profile_save.py and profiles.json added, saves profiles and returns saved profiles

JM-Lemmi commented on Client/profile_save.py:
```diff
@@ -0,0 +1,120 @@
+            except:
+                print("json error: Make sure profiles.json is formatted correctly")
+        return None
+    def set_profile(self, profile_uuid , profile_name = None, profile_color = None):
+        """
+        This method sets the profile name and/or color by the uuid
+        :param profile_uuid:
+        :param profile_name: *optional*
+        :param profile_color: *optional*
+        """
+        if self.check_file():
+            try:
+                with open(path, 'r+') as file:
+                    data = json.load(file)
+                    for profile in data:
+                        if profile["profile_uuid"] == profile_uuid:
+                            if profile_name != None:
+                                profile["profile_name"] = profile_name
+                            if profile_color != None:
+                                profile["profile_color"] = profile_color
+                            break
```
```suggestion
    def set_profile(self, profile_uuid , profile_name, profile_color):
        """
        This method sets the profile name and/or color by the uuid
        :param profile_uuid:
        :param profile_name: *optional*
        :param profile_color: *optional*
        """
        if profile_name == None || profile_color == None:
            raise ValueError("name or color cannot be none")
        
        if self.check_file():
            try:
                with open(path, 'r+') as file:
                    data = json.load(file)
                    for profile in data:
                        if profile["profile_uuid"] == profile_uuid:
                            profile["profile_name"] = profile_name
                            profile["profile_color"] = profile_color
                            break
```

JM-Lemmi commented on Client/profile_save.py:
```diff
@@ -0,0 +1,120 @@
+                entry = [{"profile_name": profile_name, "profile_uuid": profile_uuid, "profile_color": profile_color}]
+                json.dump(entry, file)
+
+#Testing
+#profile = Profile()
+#profile.add_new_profile("test", "test", "test")
+#print(profile.get_profiles())
+#print(profile.get_profile("test"))
+#profile.set_profile("test", "test2", "test3")
+#print(profile.get_profiles())
```
?

bananabr3d replied:
Ist im aktuellen durch richtige Tests

JM-Lemmi commented on json_schema/profiles.json:
```diff
@@ -0,0 +1 @@
+[]
```
bitte füllen

bananabr3d replied:
Womit? D:

JM-Lemmi replied:
na mit dem schema

Petzys replied:
Das soll gar kein Schema sein @JM-Lemmi . Die Datei ist einfach nur am falschen Ort. Siehe mein Review in #6 .

bananabr3d committed:
Update Client/profile_save.py

Co-authored-by: Julian Lemmerich <41118534+JM-Lemmi@users.noreply.github.com>

bananabr3d committed:
fix: TTTK-19 profile_save.py: merged
feat: TTT-19 profile_save.py: throws now errors if profile_color or profile_name is None, delete Functions added

bananabr3d committed:
fix: TTTK-19 profile_save.py: prints in check_file() deleted

bananabr3d committed:
feat: TTTK-19 test_profile_save.py: file created, class set up, first two tests written

bananabr3d committed:
feat: TTTK-19 test_profile.json: file created

bananabr3d committed:
feat: test_profile_save.py tests implemented

bananabr3d committed:
fix: test_profile_save.py, abs path fixed, import fixed
fix: profile_save.py: abs path added

HOOK-Hawkins commented on Client/profile_save.py:
```diff
@@ -0,0 +1,167 @@
+        """
+        This method deletes all profiles
+        """
+        if self.check_file():
```
```suggestion
        if exists(self.path):
```

bananabr3d committed:
fix: json files moved

bananabr3d committed:
fix: abs path added, test code deleted

bananabr3d committed:
fix: profile_save.py: print to error

bananabr3d committed:
fix: input and output changed to player obj

bananabr3d committed:
fix: set_profiles added, test_set_profiles added

HOOK-Hawkins commented on Client/profile_save.py:
```diff
@@ -0,0 +1,195 @@
+        This method returns all the profiles from the file
+        :return: An array of all profiles
+        """
+        if self.check_file():
```
```suggestion
        if exists(self.path):
```

HOOK-Hawkins commented on Client/profile_save.py:
```diff
@@ -0,0 +1,195 @@
+    def __init__(self, path=os.path.abspath('Client/Data/profiles.json')):
+        self.path = path
+
+    def check_file(self):
+        """
+        This method checks if the file exists
+        :return: True if the file exists, False if it does not
+        """
+        if exists(self.path):
+            return True
+        else:
+            return False
+
```
### 
```suggestion
```

HOOK-Hawkins commented on Client/profile_save.py:
```diff
@@ -0,0 +1,195 @@
+                    output.append(Player.from_dict(profile))
+                return output
+        else:
+            return None
```
```suggestion
            return []
```

HOOK-Hawkins commented on Client/profile_save.py:
```diff
@@ -0,0 +1,195 @@
+        else:
+            return None
+
+    def get_profile(self, player : Player):
+        """
+        This method returns a profile by its uuid
+        :param profile_uuid:
+        :return: profile matching given uuid
+        """
+        player_dict = player.as_dict()
+        if self.check_file():
+            try:
+                with open(self.path, 'r') as file:
+                    data = json.load(file)
+                    for profile in data:
+                        if profile["uuid"] == player_dict["uuid"]:
+                            return Player.from_dict(profile)
+            except:
+                raise RuntimeError("json error: Make sure profiles.json is formatted correctly")
+        return None
+
```
```suggestion
```

HOOK-Hawkins commented on Client/profile_save.py:
```diff
@@ -0,0 +1,195 @@
+                raise RuntimeError("json error: Make sure profiles.json is formatted correctly")
+        return None
+
+    def set_profile(self, player : Player):
+        """
+        This method sets the profile name and/or color by the uuid
+        :param profile_uuid:
+        :param profile_name:
+        :param profile_color:
+        """
+        player_dict = player.as_dict()
+        if (player_dict['display_name'] or player_dict['color']) == None:
+            raise ValueError("name or color cannot be none")
+
+        if self.check_file():
+            try:
+                with open(self.path, 'r+') as file:
+                    data = json.load(file)
+                    for profile in data:
+                        if profile["uuid"] == player_dict["uuid"]:
+                            if player_dict['display_name'] != None:
+                                profile["display_name"] = player_dict['display_name']
+                            if player_dict['color'] != None:
+                                profile["color"] = player_dict['color']
+                            break
+                    with open(self.path, 'w') as file:
+                        json.dump(data, file)
+            except:
+                raise RuntimeError("json error: Make sure profiles.json is formatted correctly")
+        return None
+
```
```suggestion
```

HOOK-Hawkins commented on Client/profile_save.py:
```diff
@@ -0,0 +1,195 @@
+                raise RuntimeError("json error: Make sure profiles.json is formatted correctly")
+        return None
+
+    def get_profile_by_name(self, player : Player):
+        if self.check_file():
+            """
+            This method returns a profile by its name
+            :param profile_name:
+            :return: profile matching given name
+            """
+            player_dict = player.as_dict()
+            try:
+                with open(self.path, 'r') as file:
+                    data = json.load(file)
+                    for profile in data:
+                        if profile["display_name"] == player_dict["display_name"]:
+                            return Player.from_dict(profile)
+            except:
+                print("json error: Make sure profiles.json is formatted correctly")
+        return None
+
```
```suggestion
```

HOOK-Hawkins commented on Client/profile_save.py:
```diff
@@ -0,0 +1,195 @@
+                print("json error: Make sure profiles.json is formatted correctly")
+        return None
+
+    def add_new_profile(self, player : Player):
+        """
+        This method adds a new profile to the file
+        :param profile_name:
+        :param profile_uuid:
+        :param profile_color:
+        """
+        if self.check_file():
+            entry = player.as_dict()
+            try:
+                with open(self.path, 'r+') as file:
+                    data = json.load(file)
+                    file.seek(0)
+                    data.append(entry)
+                    json.dump(data, file)
+                    file.truncate()
+            except:
+                raise RuntimeError("json error: Make sure profiles.json is formatted correctly")
+
+        else:
+            with open(self.path, 'w') as file:
+                entry = [player.as_dict()]
+                json.dump(entry, file)
+
```
```suggestion
```

HOOK-Hawkins commented on Client/profile_save.py:
```diff
@@ -0,0 +1,195 @@
+                entry = [player.as_dict()]
+                json.dump(entry, file)
+
+    def delete_profile(self, player : Player):
+        """
+        This method deletes a profile by its uuid
+        :param profile_uuid:
+        """
+        player_dict = player.as_dict()
+        if self.check_file():
+            try:
+                with open(self.path, 'r+') as file:
+                    data = json.load(file)
+                    for profile in data:
+                        if profile["uuid"] == player_dict["uuid"]:
+                            data.remove(profile)
+                            break
+                    else:
+                        raise ValueError(f"Profile with given uuid: {player_dict['uuid']} not found")
+                    with open(self.path, 'w') as file:
+                        json.dump(data, file)
+            except:
+                raise RuntimeError("json error: Make sure profiles.json is formatted correctly")
+
```
```suggestion
```

HOOK-Hawkins commented on Client/profile_save.py:
```diff
@@ -0,0 +1,195 @@
+            except:
+                raise RuntimeError("json error: Make sure profiles.json is formatted correctly")
+
+    def delete_profile_by_name(self, player : Player):
+        """
+        This method deletes a profile by its name
+        :param profile_name:
+        """
+        player_dict = player.as_dict()
+        if self.check_file():
+            try:
+                with open(self.path, 'r+') as file:
+                    data = json.load(file)
+                    for profile in data:
+                        if profile["display_name"] == player_dict["display_name"]:
+                            data.remove(profile)
+                            break
+                    else:
+                        raise ValueError(f"Profile with given name: {player_dict['display_name']} not found")
+                    with open(self.path, 'w') as file:
+                        json.dump(data, file)
+            except:
+                raise RuntimeError("json error: Make sure profiles.json is formatted correctly")
+
```
```suggestion
```

HOOK-Hawkins commented on Client/profile_save.py:
```diff
@@ -0,0 +1,195 @@
+        :param profile_color:
+        """
+
+        if self.check_file():
```
```suggestion
```

HOOK-Hawkins commented on Client/profile_save.py:
```diff
@@ -0,0 +1,195 @@
+                    json.dump(entry, file)
+            except:
+                raise RuntimeError("json error: Make sure profiles.json is formatted correctly")
+        return None
```
```suggestion
```

bananabr3d committed:
Update Client/profile_save.py

Co-authored-by: Hauke Platte <53154723+HOOK-Hawkins@users.noreply.github.com>

bananabr3d committed:
Update Client/profile_save.py

Co-authored-by: Hauke Platte <53154723+HOOK-Hawkins@users.noreply.github.com>

bananabr3d committed:
Update Client/profile_save.py

Co-authored-by: Hauke Platte <53154723+HOOK-Hawkins@users.noreply.github.com>

bananabr3d committed:
Update Client/profile_save.py

Co-authored-by: Hauke Platte <53154723+HOOK-Hawkins@users.noreply.github.com>

bananabr3d committed:
Update Client/profile_save.py

Co-authored-by: Hauke Platte <53154723+HOOK-Hawkins@users.noreply.github.com>

bananabr3d committed:
Update Client/profile_save.py

Co-authored-by: Hauke Platte <53154723+HOOK-Hawkins@users.noreply.github.com>

bananabr3d committed:
Update Client/profile_save.py

Co-authored-by: Hauke Platte <53154723+HOOK-Hawkins@users.noreply.github.com>

bananabr3d committed:
Update Client/profile_save.py

Co-authored-by: Hauke Platte <53154723+HOOK-Hawkins@users.noreply.github.com>

bananabr3d committed:
Update Client/profile_save.py

Co-authored-by: Hauke Platte <53154723+HOOK-Hawkins@users.noreply.github.com>

bananabr3d committed:
Update Client/profile_save.py

Co-authored-by: Hauke Platte <53154723+HOOK-Hawkins@users.noreply.github.com>

bananabr3d committed:
fix: useless functions deleted, indentation error fixed

bananabr3d committed:
fix: test_profile_save.py: tests adjusted

bananabr3d committed:
fix: test_profile_save.py: deleted file case implemented
